api/client/usermanager/client_test.go

```diff
@@ -4,170 +4,255 @@
+	_, _, err := client.AddUser("foobar", "Foo Bar", "password")
 	c.Assert(err, gc.ErrorMatches, "failed to create user: username unavailable")
 }
 
 func (s *usermanagerSuite) TestAddUserResponseError(c *gc.C) {
-	usermanager.PatchResponses(s, s.usermanager,
-		func(interface{}) error {
-			return errors.New("call error")
-		},
-	)
-	_, _, err := s.usermanager.AddUser("foobar", "Foo Bar", "password")
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	args := params.AddUsers{
+		Users: []params.AddUser{{Username: "foobar", DisplayName: "Foo Bar", Password: "password"}},
+	}
+
+	result := new(params.AddUserResults)
+	results := params.AddUserResults{
+		Results: make([]params.AddUserResult, len(args.Users)),
+	}
+	mockFacadeCaller := mocks.NewMockFacadeCaller(ctrl)
+	mockFacadeCaller.EXPECT().FacadeCall("AddUser", args, result).SetArg(2, results).Return(errors.New("call error"))
+
+	client := usermanager.NewClientFromCaller(mockFacadeCaller)
+	_, _, err := client.AddUser("foobar", "Foo Bar", "password")
 	c.Assert(err, gc.ErrorMatches, "call error")
 }
 
 func (s *usermanagerSuite) TestAddUserResultCount(c *gc.C) {
-	usermanager.PatchResponses(s, s.usermanager,
-		func(result interface{}) error {
-			if result, ok := result.(*params.AddUserResults); ok {
-				result.Results = make([]params.AddUserResult, 2)
-				return nil
-			}
-			return errors.New("wrong result type")
-		},
-	)
-	_, _, err := s.usermanager.AddUser("foobar", "Foo Bar", "password")
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	args := params.AddUsers{
+		Users: []params.AddUser{{Username: "foobar", DisplayName: "Foo Bar", Password: "password"}},
+	}
+
+	result := new(params.AddUserResults)
+	results := params.AddUserResults{
+		Results: make([]params.AddUserResult, 2),
+	}
+	mockFacadeCaller := mocks.NewMockFacadeCaller(ctrl)
+	mockFacadeCaller.EXPECT().FacadeCall("AddUser", args, result).SetArg(2, results).Return(nil)
+
+	client := usermanager.NewClientFromCaller(mockFacadeCaller)
+	_, _, err := client.AddUser("foobar", "Foo Bar", "password")
 	c.Assert(err, gc.ErrorMatches, "expected 1 result, got 2")
 }
 
 func (s *usermanagerSuite) TestRemoveUser(c *gc.C) {
-	tag, _, err := s.usermanager.AddUser("jjam", "Jimmy Jam", "password")
-	c.Assert(err, jc.ErrorIsNil)
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
 
-	// Ensure the user exists.
-	user, err := s.State.User(tag)
+	args := params.AddUsers{
+		Users: []params.AddUser{{Username: "jjam", DisplayName: "Jimmy Jam", Password: "password"}},
+	}
+
+	result := new(params.AddUserResults)
+	results := params.AddUserResults{
+		[]params.AddUserResult{{Tag: "user-jjam"}},
+	}
+	mockFacadeCaller := mocks.NewMockFacadeCaller(ctrl)
+	mockFacadeCaller.EXPECT().FacadeCall("AddUser", args, result).SetArg(2, results).Return(nil)
+
+	client := usermanager.NewClientFromCaller(mockFacadeCaller)
+	tag, _, err := client.AddUser("jjam", "Jimmy Jam", "password")
```

Comment:
This code is not required. It was used before to setup the remove user part of the test, but not necessary with the mocks.